CrowsZA/functions/fn_fireSupport.sqf

```diff
@@ -0,0 +1,205 @@
+/*/////////////////////////////////////////////////
+Author: Radkewolf
+			   
+File: fn_fireSupport.sqf
+Parameters: pos
+Return: none
+
+Sets off firesupport in the targeted area
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3], ["_logic",objNull,[objNull]]];
+
+// Validate module target
+scopeName "Main";
+private _fnc_errorAndClose = {
+    params ["_msg"];
+    diag_log _msg;
+    breakOut "Main";
+};
+
+// open dialog
+//ZEN
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+	_dialogResult params
+	[
+		"_type",
+		"_customType",
+		"_radius",
+		"_seconds",
+		"_delay",
+		"_salvos",
+		"_guns"
+	];
+
+	//Get in params again
+	_in params [["_pos",[0,0,0],[[]],3], ["_logic",objNull,[objNull]]];
+
+	//Check if placed on an object
+	_create = isNull _logic;
+
+	//Check if object is firesupport module
+	if(!_create)
+	then
+	{
+		if(!(_logic isKindOf "Logic"))
+		then
+		{
+			_create = true;
+		}
+		else {
+			_mod = _logic getVariable ["_module", ""];
+			if(_mod != "firesupport")
+			then
+			{
+				_create = true;
+			};
+		}
+	};
+
+	//Create new module if not placed on existing one
+	if(_create)
+	then{
+		_logicCenter = createCenter sideLogic;
+		_logicGroup = createGroup _logicCenter;
+		_logic = _logicGroup createUnit ["Logic", _pos, [], 0, "NONE"];
+		_logic setVariable ["_module", "firesupport"];
+		["zen_common_addObjects", [[_logic], objNull]] call CBA_fnc_serverEvent;
+	};
+
+	//Use specific cfgAmmo if filled
+	if(_customType != "")
+	then{
+		_type = _customType;
+	};
+	diag_log "guns:";
+	diag_log str _guns;
+	//Set variables to logic gameobject
+	_logic setVariable ["_type", _type];
+	_logic setVariable ["_radius", _radius];
+	_logic setVariable ["_seconds", _seconds];
+	_logic setVariable ["_salvos", round _salvos];
+	_logic setVariable ["_guns", round _guns];
+	
+
+	_spawnBarrage = 
+	{
+		_logic = (_this select 0);
+		_code = (_this select 1);
+
+		if(!(isNull _logic))
+		then{
+			//Read values from logic gameobject
+			_pos = getPos _logic;
+			_type = _logic getVariable "_type";
+			_radius = _logic getVariable "_radius";
+			_seconds = _logic getVariable "_seconds";
+			_salvos = _logic getVariable "_salvos";
+			_guns = _logic getVariable "_guns";
+
+			//Spawn salvo with values
+			[_pos, _type, _radius, _guns, [0, 0.5]] spawn BIS_fnc_fireSupportVirtual;
+
+			//Wait till next salvo
+			sleep _seconds;
+			
+			//Check if more salvos should be spawned
+			if (_salvos > 1) 
+			then { 
+				//Reduces amount of remaining salvos and spawn next
+				_logic setVariable ["_salvos", _salvos -1];
+				[_logic, _code] spawn _code; 
+			} 
+			else {if (_salvos == 1) 
+				then{
+					//Delete if all salvos are done
+					deleteVehicle _logic;
+				}
+				else {if(_salvos == 0)
+					then {
+						//Don't stop if savlos is equal to 0
+						[_logic, _code] spawn _code;
+					}
+					else{
+						deleteVehicle _logic;
+					}
+				}
+			}
```

Comment:
Instead of nested if'sentences like this, you can use switch, which in my opinion is more readable. It's a quite handy trick to use switch statements with ``true`` and have each case use if syntax to test.
```sqf
switch (true) do {
	// case more than 1 salvo left
	case (_salvos > 1): {
		//Reduces amount of remaining salvos and spawn next
		_logic setVariable ["_salvos", _salvos -1];
		[_logic, _code] spawn _code; 
	};

	// case salvo == 1
	case (_salvos == 1): {
		//Delete if all salvos are done
		deleteVehicle _logic;
	};

	// case salvos == 0
	case (_salvos == 0): {
		[_logic, _code] spawn _code;
	};
	
	// default case 
	default {
		// I think you had this as a, if nothing else match, we just delete it clause
		deleteVehicle _logic;
	};
};
```

Reply:
I didn't know you could do that. That's pretty nice. I'll give it a go.